### PR TITLE
update readme and rules for 3.12.0 release

### PR DESCRIPTION
**Pull request by Madhu-1**
Updating readme and Rules for the 3.12.0 release

Depends-on: #4766

### CODE REVIEW

**Comment by Madhu-1** [comment]
Running 1.31 tests on https://github.com/ceph/ceph-csi/pull/4757

**Review by nixpanic — DISMISSED** [review]

File: .mergify.yml [hunk 1]
```diff
@@ -57,12 +57,15 @@ queue_rules:
               - "status-success=ci/centos/k8s-e2e-external-storage/1.26"
               - "status-success=ci/centos/k8s-e2e-external-storage/1.27"
               - "status-success=ci/centos/k8s-e2e-external-storage/1.28"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.29"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.25"
```
Comment (Rakshith-R) [inline .mergify.yml]:
```suggestion
```

**Review by Rakshith-R — COMMENTED** [review]
Tiny nit,
k8s 1.25 test was added for v3.11 branch at one place

**Review by Rakshith-R — APPROVED** [review]

**Review by iPraveenParihar — APPROVED** [review]
Thanks!

**Comment by Madhu-1** [comment]
Addinng DNM to avoid merging , started CI job for 1.31

**Comment by nixpanic** [comment]
> Running 1.31 tests on #4757

This one failed: https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/k8s-e2e-external-storage-1.31/detail/k8s-e2e-external-storage-1.31/2/pipeline

**Comment by Madhu-1** [comment]
```
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f ceph-csi $]./scripts/minikube.sh up
minikube already installed with v1.31.0
😄  minikube v1.31.0 on Fedora 40
    ▪ KUBECONFIG=/home/mrajanna/workspace/src/github.com/k8s-cluster/kubeconfig
    ▪ MINIKUBE_VERSION=v1.31.0
❗  minikube skips various validations when --force is supplied; this may lead to unexpected behavior
❗  Specified Kubernetes version 1.31.0 is newer than the newest supported version: v1.27.3. Use `minikube config defaults kubernetes-version` for details.
✨  Using the kvm2 driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating kvm2 VM (CPUs=12, Memory=4096MB, Disk=32768MB) ...
🐳  Preparing Kubernetes v1.31.0 on Docker 24.0.4 ...
    ▪ kubelet.resolv-conf=/run/systemd/resolve/resolv.conf
    ▪ kubelet.read-only-port=10255
❌  Unable to load cached images: loading cached images: stat /home/mrajanna/.minikube/cache/images/amd64/registry.k8s.io/pause_test2: no such file or directory
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
E0814 13:57:27.599459  171608 pod_ready.go:66] WaitExtra: waitPodCondition: error getting pod "coredns-6f6b679f8f-967t5" in "kube-system" namespace (skipping!): pods "coredns-6f6b679f8f-967t5" not found
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
root        2117    2096 27 11:57 ?        00:00:05 kube-apiserver --advertise-address=192.168.39.84 --allow-privileged=true --authorization-mode=Node,RBAC --client-ca-file=/var/lib/minikube/certs/ca.crt --enable-admission-plugins=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota --enable-bootstrap-token-auth=true --etcd-cafile=/var/lib/minikube/certs/etcd/ca.crt --etcd-certfile=/var/lib/minikube/certs/apiserver-etcd-client.crt --etcd-keyfile=/var/lib/minikube/certs/apiserver-etcd-client.key --etcd-servers=https://127.0.0.1:2379 --kubelet-client-certificate=/var/lib/minikube/certs/apiserver-kubelet-client.crt --kubelet-client-key=/var/lib/minikube/certs/apiserver-kubelet-client.key --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname --proxy-client-cert-file=/var/lib/minikube/certs/front-proxy-client.crt --proxy-client-key-file=/var/lib/minikube/certs/front-proxy-client.key --requestheader-allowed-names=front-proxy-client --requestheader-client-ca-file=/var/lib/minikube/certs/front-proxy-ca.crt --requestheader-extra-headers-prefix=X-Remote-Extra- --requestheader-group-headers=X-Remote-Group --requestheader-username-headers=X-Remote-User --secure-port=8443 --service-account-issuer=https://kubernetes.default.svc.cluster.local --service-account-key-file=/var/lib/minikube/certs/sa.pub --service-account-signing-key-file=/var/lib/minikube/certs/sa.key --service-cluster-ip-range=10.96.0.0/12 --tls-cert-file=/var/lib/minikube/certs/apiserver.crt --tls-private-key-file=/var/lib/minikube/certs/apiserver.key
root        3843       1  0 11:57 ?        00:00:00 /var/lib/minikube/binaries/v1.31.0/kubelet --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --config=/var/lib/kubelet/config.yaml --container-runtime-endpoint=unix:///var/run/cri-dockerd.sock --hostname-override=minikube --kubeconfig=/etc/kubernetes/kubelet.conf --node-ip=192.168.39.84 --read-only-port=10255 --resolv-conf=/run/systemd/resolve/resolv.conf --v=4
docker      4012    4004  0 11:57 pts/0    00:00:00 bash -c ps -Af |grep kubelet
docker      4014    4012  0 11:57 pts/0    00:00:00 grep kubelet
* "The 'default-storageclass' addon is disabled
* "The 'storage-provisioner' addon is disabled
    > kubectl.sha256:  64 B / 64 B [-------------------------] 100.00% ? p/s 0s
    > kubectl:  53.77 MiB / 53.77 MiB [-------------] 100.00% 6.26 MiB p/s 8.8s
Kubernetes control plane is running at https://192.168.39.84:8443
CoreDNS is running at https://192.168.39.84:8443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f ceph-csi $]kubectl version
Client Version: v1.30.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.31.0

```

it worked locally, lets try one more time 🗡️

**Comment by Madhu-1** [comment]
https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/k8s-e2e-external-storage-1.31/detail/k8s-e2e-external-storage-1.31/3/pipeline/107/ 1.31 passed

**Comment by Madhu-1** [comment]
Removing DNM as all the required tests are passed here https://github.com/ceph/ceph-csi/pull/4766

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
@mergifyio queue

**Comment by Madhu-1** [comment]
@nixpanic PTAL

**Comment by Madhu-1** [comment]
I think this needs manual merge

**Comment by nixpanic** [comment]
> I think this needs manual merge

It should not, Mergify can merge changes to its own configuration file.

**Pull request merged**